Sources/Offline/OfflineManager.swift

```diff
@@ -0,0 +1,285 @@
+                DispatchQueue.main.async {
+                    PKLog.error("Load item metadata failed: " + error.localizedDescription)
+                    callback(OfflineManagerError.loadItemMetadataFailed(message: error.localizedDescription), nil)
+                }
+            }
+        }
+    }
+    
+    /**
+        Start or resume downloading the asset.
+    
+        * Parameters:
+            * assetId: The asset's id.
+    */
+    public func startAssetDownload(assetId: String) {
+        do {
+            try ContentManager.shared.startItem(id: assetId)
+        } catch {
+            PKLog.error(error.localizedDescription)
+        }
+    }
+    
+    /**
+        Pause downloading the asset.
+    
+        * Parameters:
+            * assetId: The asset's id.
+    */
+    public func pauseAssetDownload(assetId: String) {
+        do {
+            try ContentManager.shared.pauseItem(id: assetId)
+        } catch {
+            PKLog.error(error.localizedDescription)
+        }
+    }
+    
+    /**
+        Remove the asset with all it's data.
+    
+        * Parameters:
+            * assetId: The asset's id.
+    */
+    public func removeAssetDownload(assetId: String) {
+        do {
+            guard let url = try ContentManager.shared.itemPlaybackUrl(id: assetId) else {
+                PKLog.error("Can't get the local url in order to remove the downloaded asset.")
+                return
+            }
+            
+            localAssetsManager.unregisterDownloadedAsset(location: url, callback: { (error) in
+                PKLog.debug("Unregister complete.")
+            })
+            
+            try? ContentManager.shared.removeItem(id: assetId)
+            
+        } catch {
+            PKLog.error(error.localizedDescription)
+        }
+    }
+    
+    public func getAssetInfo(assetId: String) -> AssetInfo? {
+        guard let dtgItem = try? ContentManager.shared.itemById(assetId) else { return nil }
+        
+        return AssetInfo(item: dtgItem)
+    }
+    
+    public func getLocalPlaybackEntry(assetId: String) -> PKMediaEntry? {
+        guard let playbackURL = try? ContentManager.shared.itemPlaybackUrl(id: assetId) else {
+            PKLog.debug("Can't get local url for \(assetId)")
+            return nil
+        }
+        
+        return localAssetsManager.createLocalMediaEntry(for: assetId, localURL: playbackURL)
+    }
+}
+    
+// MARK: - DRM
+
+extension OfflineManager {
+    
+    public func getDRMStatus(assetId: String) -> DRMStatus? {
+        guard let url = try? ContentManager.shared.itemPlaybackUrl(id: assetId) else {
+            PKLog.debug("Can't get local url for \(assetId)")
+            return nil
+        }
+        
+        guard let fpsExpirationInfo = localAssetsManager.getLicenseExpirationInfo(location: url) else { return nil }
+        
+        return DRMStatus(fpsExpirationInfo)
+    }
+    
+    public func renewDrmAssetLicense(mediaEntry: PKMediaEntry) {
```

Comment:
I think it should be "Asset DRM License".
Anyway, use either Drm or DRM in all places. For example, above you have `getDRMStatus?()`.

Reply:
Fixed to renewAssetDRMLicense.